conans/client/tools/net.py

```diff
@@ -49,7 +52,33 @@ def ftp_download(ip, filename, login='', password=''):
 
 def download(url, filename, verify=True, out=None, retry=None, retry_wait=None, overwrite=False,
              auth=None, headers=None, requester=None, md5='', sha1='', sha256=''):
+    """Retrieves a file from a given URL into a file with a given filename.
+       It uses certificates from a list of known verifiers for https downloads,
+       but this can be optionally disabled.
 
+    :param url: URL to download. It can be a list, which only the first one will be downloaded, and
+                the follow URLs will be used as mirror in case of download error.
+    :param filename: Name of the file to be created in the local storage
+    :param verify: When False, disables https certificate validation
+    :param out: An object with a write() method can be passed to get the output. stdout will use if
+                not specified
+    :param retry: Number of retries in case of failure. Default is overriden by general.retry in the
+                  conan.conf file or an env variable CONAN_RETRY
+    :param retry_wait: Seconds to wait between download attempts. Default is overriden by
+                       general.retry_wait in the conan.conf file or an env variable CONAN_RETRY_WAIT
+    :param overwrite: When True, Conan will overwrite the destination file if exists. Otherwise it
+                      will raise an exception
+    :param auth: A tuple of user and password to use HTTPBasic authentication
+    :param headers: A dictionary with additional headers
+    :param requester: HTTP requests instance
+    :param md5: MD5 hash code to check the downloaded file. It can be a list, where each index
```

Comment:
I don't get this, if they are mirrors, they are mirrors, the file should be the same!
This reads a bit problematic for me.

Reply:
Yes, but no. Here we have an example where it's not true: https://github.com/conan-io/conan-center-index/pull/1085/files/fb7b316a1dca315bb78d386a17013e54286b1444#diff-0fb301696473b77508534bbd398b29bc

Reply:
It's not the same time that I see a mirror with different checksum. I can remember some cases where Bincrafters used to download from author's server and moved to github, where the exactly same version is available, but the release file diverges due a generated file, like date and time.

Reply:
Okay, I agree. Only the URL list. Filename and checksum should be the same for all downloads. We have situations where it won't work, but we should consider as an upstream "error".

Reply:
Yes, I agree. Having different checksums it is too bad to be a built-in feature, should be an exception, and not something that ConanCenter would accept.